### ISSUE TITLE
Hotspot mode not working for devices with Halium 9 or higher

### ISSUE THREAD

**Issue by Flohack74** [ubports/ubuntu-touch#1939]
- Device: axolotl
- Channel: devel
- Build: any since ever

### Steps to reproduce
- Turn hotspot on
- UI times out
- No hotspot activated
- WiFi also is dead needs reboot

### Expected behavior
Hotspot switches on and can be used

### Logfiles and additional information
This is due to Canonical implementing a weird firmware switcher in wpa_supplicant, where via DBus a vertain call to Android´s libwifi is being made. Since more modern Android versions do not have that lib anymore the whole functionality is broken now.

**Comment by Vistaus** [comment]
Same on the Cosmo Communicator. Reported it months ago with no response. Is there a workaround for this, like downgrading wpa_supplicant or something?

**Comment by Flohack74** [comment]
No, sorry, no workaround, it needs proper fixing to use the Android API

**Comment by buchnerh** [comment]
Is it really a Halium-9-problem? My Vollaphone should also be a Halium-9-device but here everything is working fine.

**Comment by Flohack74** [comment]
@buchnerh for Vollaphone this was fixed with a hack to work, also because its Mediatek it only works there.

**Comment by j2g2rp** [comment]
> @buchnerh for Vollaphone this was fixed with a hack to work, also because its Mediatek it only works there.

Talking about vollaphone... On Vollaphone X sometimes hotspot doesn't work and a reboot is needed. Is related?

**Comment by Flohack74** [comment]
I can be related bu who knows. needs further digging probably.
